src/libsyntax_pos/symbol.rs

```diff
@@ -1137,3 +1138,20 @@ impl fmt::Display for SymbolStr {
         fmt::Display::fmt(self.string, f)
     }
 }
+
+impl<CTX> HashStable<CTX> for SymbolStr {
+    #[inline]
+    fn hash_stable(&self, hcx: &mut CTX, hasher: &mut StableHasher) {
+        let str = self as &str;
```

Comment:
I wasn't aware `as` conversions applied derefs. Anyway I think this would be cleaner if it just did `self.string.hash_stable(hcx, hasher)`